implement input, textarea, and floating label

jefrog1844 opened the pull request:
Let me know if there are any questions or anything you'd like changed.

Jeff

amorey commented:
This is awesome! Congrats! Some comments/questions:

- Some of the new files have executable filesystem permissions (e.g. form-page.component.ts, input.component.ts). Can you make them unexecutable?
- Should the Input and Textarea be split into two separate mui-angular secondary endpoints?
- What is the purpose of the FormModule? For the `muiFloatingLabel` directive? Is it possible to add the directive to the InputModule and TextareaModule instead?
- What do you think about changing the name of `muiFloatingLabel` to `float-label` or something else?
- What do you think about using `placeholder` instead of `hint`? I used `placeholder` in MUI React and I think that Angular Material does the same.

jefrog1844 commented:
I agree with your changes and make updates in the next evening or two.

As for the floating label directive, Angular recommends sharing modules (
https://angular.io/guide/sharing-ngmodules) which simplifies things.  I can
create a shared folder that can hold directives or anything else that needs
shared.  All that would be required is that 'app.module.ts' import either
InputModule or TextareaModule and it would automatically provide the
float-label directive.

Jeff

On Tue, Sep 29, 2020 at 2:54 AM Andres Morey <notifications@github.com>
wrote:

> This is awesome! Congrats! Some comments/questions:
>
>    - Some of the new files have executable filesystem permissions (e.g.
>    form-page.component.ts, input.component.ts). Can you make them unexecutable?
>    - Should the Input and Textarea be split into two separate mui-angular
>    secondary endpoints?
>    - What is the purpose of the FormModule? For the muiFloatingLabel
>    directive? Is it possible to add the directive to the InputModule and
>    TextareaModule instead?
>    - What do you think about changing the name of muiFloatingLabel to
>    float-label or something else?
>    - What do you think about using placeholder instead of hint? I used
>    placeholder in MUI React and I think that Angular Material does the
>    same.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/muicss/mui-angular/pull/5#issuecomment-700519392>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM6FHO37KJFWHOEIVLAVXQLSIGHCPANCNFSM4R5F4DJA>
> .
>

jefrog1844 commented:
- updated file permissions to remove executable permissions
- split textarea to its own entry point
- removed form entry point
- moved floating label directive to a shared module and renamed as floatLabel to follow camel case angular recommendation for directives
- rename hint as placeholder
